static_frame/core/frame.py

```diff
@@ -7378,6 +7378,35 @@ def sample(self,
 
     #---------------------------------------------------------------------------
 
+    def _container_from_index_values(
+            self,
+            values: np.ndarray[tp.Any, tp.Any],
+            axis: int,
+            ) -> TSeriesAny:
+
+        if values.ndim == 1:
+            post = values
+        else:
+            post = np.fromiter(
+                map(tuple, values),
+                count=len(values),
+                dtype=object,
+            )
+            post.flags.writeable = False
```

Comment:
I think you can use `arraykit.array_to_tuple_array` for an optimized version of this move.